airflow/models.py

```diff
@@ -686,8 +686,8 @@ def parse_from_uri(self, uri):
         self.conn_type = conn_type
         self.host = hostname
         self.schema = temp_uri.path[1:]
-        self.login = temp_uri.username
-        self.password = temp_uri.password
+        self.login = unquote(temp_uri.username or '')
+        self.password = unquote(temp_uri.password or '')
```

Comment:
Can you add an explicit test that passing a URI-quoted uri in works as expected please